onnxruntime/core/optimizer/skip_layer_norm_fusion.cc

```diff
@@ -57,13 +57,18 @@ static bool CheckFirstAdd(Node& add, ProviderType providertype) {
 // Add2 is the 2nd add of the to be fused sub-graph
 // The 1st input should be a 3D tensor
 // The 2nd input should be a 1D constant value
-static bool CheckSecondAdd(Node& add, ProviderType providertype) {
+static bool CheckSecondAdd(Graph& graph, Node& add, ProviderType providertype) {
   if (providertype != add.GetExecutionProviderType() ||
       !IsSupportedDataType(add) ||
       add.GetOutputEdgesCount() != 1) {
     return false;
   }
 
+  // The 2nd input should be a constant value
+  if (!graph_utils::NodeArgIsConstant(graph, *(add.MutableInputDefs()[1]))) {
```

Comment:
Do we need check the input shape for this?

Reply:
yes, just under the constant check